### DISCUSSION TITLE
Does the Pattern Editor has autosave function?

### DISCUSSION THREAD

**Discussion by xlbao123** [WerWolv/ImHex#1765] [Pattern Language Q&A] [answered]
I create a project, and write some code in pattern editor. Unfortunally the editor crashed, after I restart the ImHex, the pattern code looked missed?

**Comment by paxcut** [comment, marked as answer]
there is an auto save project in the settings that lets you select how often the current session is saved, but there is no auto save for pattern files only. You can also save the current session from the editor using alt-s.
